CtrlVAF/CtrlVAF.Core/Core/ConfigurableVaultApplicationBase.cs

```diff
@@ -0,0 +1,93 @@
+
+using MFilesAPI;
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+
+namespace CtrlVAF.Core
+{
+    public abstract partial class ConfigurableVaultApplicationBase<TSecureConfiguration> :
+        MFiles.VAF.Extensions.MultiServerMode.ConfigurableVaultApplicationBase<TSecureConfiguration> where TSecureConfiguration : class, new()
+    {
+        public TaskQueueBackgroundOperationManager TaskQueueBackgroundOperationManager { get; set; }
+
+        public RecurringBackgroundOperations PermanentBackgroundOperations { get; }
+             = new RecurringBackgroundOperations();
+
+        public OnDemandBackgroundOperations OnDemandBackgroundOperations { get; }
+             = new OnDemandBackgroundOperations();
+
+        public Dispatcher BackgroundDispatcher { get; private set; }
+
+        public Dispatcher EventDispatcher { get; private set; }
+
+        public Dispatcher<IEnumerable<ValidationFinding>> ValidatorDispatcher { get; private set; }
+
+        internal TSecureConfiguration GetConfig()
+        {
+            return Configuration;
+        }
+
+        public override void StartOperations(Vault vaultPersistent)
+        {
+            BackgroundDispatcher = new BackgroundDispatcher<TSecureConfiguration>(this);
+
+            EventDispatcher = new EventDispatcher();
+
+            ValidatorDispatcher = new ValidatorDispatcher();
+
+            if (this.GetType().IsDefined(typeof(UseLicensingAttribute)))
+            {
+                var content = License?.Content<LicenseContentBase>();
+
+                BackgroundDispatcher = new LicensedDispatcher(BackgroundDispatcher, content);
+
+                EventDispatcher = new LicensedDispatcher(EventDispatcher, content);
+
+                ValidatorDispatcher = new LicensedDispatcher<IEnumerable<ValidationFinding>>(ValidatorDispatcher, content);
+            }
+
+            base.StartOperations(vaultPersistent);
+        }
+
+        /// <summary>
+        /// Registers background operations
+        /// </summary>
+        protected override void StartApplication()
+        {
+            TaskQueueBackgroundOperationManager = new TaskQueueBackgroundOperationManager(
+                this,
+                this.GetType().FullName.Replace(".", "-") + " - BackgroundOperations"
+                );
+
+            try
+            {
+                BackgroundDispatcher.Dispatch();
+            }
+            catch(Exception e)
+            {
+                SysUtils.ReportErrorMessageToEventLog("Could not dispatch the background operations.", e);
+                return;
+            }
+
+            base.StartApplication();
+        }
+
+        protected override IEnumerable<ValidationFinding> CustomValidation(Vault vault, TSecureConfiguration config)
+        {
+            var command = new ValidatorCommand<TSecureConfiguration> { Vault = vault, Configuration = config };
+
+            return ValidatorDispatcher.Dispatch(command);
```

Comment:
Should the base validation method be called as well, or is that not necessary?

Reply:
I don't think so but I'll add it to be sure.

Reply:
I was just thinking if the base method also did some validation that you would want to be returned. For instance, in the example on the developer portal, the base method is called [https://developer.m-files.com/Frameworks/Vault-Application-Framework/Configuration/Custom-Validation/](url)

The things returned by the base call could just be concattenated to the once retrieved by the validatordispatcher and then returned in the end.

But again, I am unsure if the base call returns any findings of value.

Reply:
I don't think it does. But I'll just concatenate them then.